src/transformStateWithClones.js

```diff
@@ -7,7 +7,29 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateLogs = [];
+  let stateCopy = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(stateCopy, action.extraData);
+        break;
+      case 'removeProperties':
+        for (const entriesToRemove of action.keysToRemove) {
+          delete stateCopy[entriesToRemove];
+        }
+        break;
+      case 'clear':
+        stateCopy = {};
+        break;
+      default:
+        break;
```

Comment:
Why did you leave it empty? The `default` should throw an error or return a string "Something went wrong" or smth like that